File: Java/169.Majority_Element.java
class Solution {
    public int majorityElement(int[] nums) {
        Integer maj = null;
        int count = 0;
        for (int num : nums) {
            if (count == 0) {
                count = 1;
                maj = num;
            }
            else if (num == maj) {
                count += 1;
            }
            else {
                count -= 1;
            }
        }       
        return maj;
    }
}

File: Python3/169.Majority_Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        maj = None
        count = 0
        for num in nums:
            if count == 0:
                maj = num
                count = 1
            elif num == maj:
                count += 1
            else:
                count -= 1
        return maj